Behaviors/FrameworkElementBehaviors.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Data;
+using Thingie.WPF.Converters;
+
+namespace Thingie.WPF.Behaviors
+{
+    /// <summary>
+    /// Represents behaviors that can be applied to a <see cref="FrameworkElement"/>.
+    /// </summary>
+    public class FrameworkElementBehaviors
+    {
+        /// <summary>
+        /// A <see cref="DependencyProperty"/> used to store the element's <see cref="FrameworkElement"/> children that impact the element's <see cref="Visibility"/>. This provides a safe and controlled way to dynamically set the element's properties based on the registered children's properties.
+        /// </summary>
+        public static readonly DependencyProperty VisibilityChildrenDependenciesProperty = 
+            DependencyProperty.RegisterAttached("VisibilityChildrenDependencies", typeof(List<FrameworkElement>), typeof(FrameworkElementBehaviors), new PropertyMetadata(null, OnVisibilityChildrenDependenciesChanged));
+
+        /// <summary>
+        /// Adds the specified <see cref="FrameworkElement"/> value to the element's <see cref="VisibilityChildrenDependenciesProperty"/>.
+        /// </summary>
+        /// <param name="element">The element to whose <see cref="VisibilityChildrenDependenciesProperty"/> the <see cref="FrameworkElement"/> value will be added.</param>
+        /// <param name="value">The <see cref="FrameworkElement"/> to be added to the <see cref="VisibilityChildrenDependenciesProperty"/>.</param>
+        public static void AddVisibilityChildDependency(FrameworkElement element, FrameworkElement value)
```

Comment:
Why not just AddVisibilityChild?

Reply:
I wanted to state the fact that the element _**depends**_ on its children, because in a generic way the children depend on the parent

Reply:
What confuses me is the naming similarity to dependency properties in WPF. Is that intentional or is this a different kind of dependency?

Reply:
I'd steer clear of the dependency word, e.g. `DependencyProperty VisibilityChildrenDependenciesProperty` seems to have two different meanings of the word dependency.

Reply:
This is all nitpicking, though